internal/internal_decision_state_machine.go

```diff
@@ -905,6 +991,8 @@ func (h *commandsHelper) recordLocalActivityMarker(activityID string, details ma
 	// to correctly incrementing counter before adding the command.
 	h.getNextID()
 	h.addCommand(command)
+	// LA markers should be the first command executed
+	h.orderedCommands.MoveToFront(h.commands[command.getID()])
```

Comment:
@samarabbas So, this line makes that unit test referenced in the other comment pass.

However, I'm not sure that what it's asserting is actually a real concern. To check this, I wrote the workflow here: https://github.com/temporalio/sdk-go/pull/302/files#diff-8118e44ac15fb5ab2e2ece2ab52fd23062a10b5cc9c55ee5b3620a1242cff27cR749

Then I ran it against `master`, and against my changes. Without my changes, `TimerCanceled` shows up in the history *before* `MarkerRecorded` -- even though the timer is canceled after the local activity is started. After my changes the order is more like what the unit test claims, but in either case the workflows complete just fine.

Any thoughts on this? My gut feel is the unit test maybe isn't actually asserting anything important by saying the marker must be recorded before the timer is cancelled (in the same workflow task), in which case we can get rid of this line and there's no more reshuffling.